java/bulk-export/bulk-export-core/src/main/java/sleeper/bulkexport/model/BulkExportQuery.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright 2022-2024 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package sleeper.bulkexport.model;
+
+import java.util.Objects;
+import java.util.UUID;
+
+/**
+ * A request for a bulk export on a given table.
+ */
+public class BulkExportQuery {
+    private final String exportId;
+    private final String tableId;
+    private final String tableName;
+
+    private BulkExportQuery(Builder builder) {
+        exportId = builder.exportId != null ? builder.exportId : UUID.randomUUID().toString();
+
+        if (builder.tableId == null && builder.tableName == null) {
+            throw new BulkExportQueryValidationException(
+                    builder.exportId, "tableId or tableName field must be provided");
+        } else if (builder.tableId != null && builder.tableName != null) {
+            throw new BulkExportQueryValidationException(
+                    builder.exportId,
+                    "tableId or tableName field must be provided, not both");
+        } else {
+            tableId = builder.tableId;
+            tableName = builder.tableName;
+        }
+    }
+
+    /**
+     * Checks that the object created is valid and has all of the required fields.
+     *
+     * @return the validated object.
+     */
+    public BulkExportQuery validate() {
+        return BulkExportQuery.builder()
+                .tableId(tableId)
```

Comment:
I like how you've done the validation, we could think about doing this elsewhere.